caput/tests/test_lint.py

```diff
@@ -0,0 +1,98 @@
+# === Start Python 2/3 compatibility
+from __future__ import absolute_import, division, print_function, unicode_literals
+from future.builtins import *  # noqa  pylint: disable=W0401, W0614
+from future.builtins.disabled import *  # noqa  pylint: disable=W0401, W0614
+
+# === End Python 2/3 compatibility
+
+import pytest
+import tempfile
+import yaml
+
+from click.testing import CliRunner
+
+from caput.config import Property
+from caput.pipeline import TaskBase
+from caput.scripts import runner as caput_script
+
+
+class DoNothing(TaskBase):
+    pass
+
+
+class DoNothing2(DoNothing):
+    a_list = Property(proptype=list)
+
+
+@pytest.fixture()
+def simple_config():
+    yield {
+        "pipeline": {
+            "tasks": [
+                {
+                    "type": "caput.tests.test_lint.DoNothing",
+                    "out": "out1",
+                },
+                {
+                    "type": "caput.tests.test_lint.DoNothing",
+                    "out": "out2",
+                    "in": "out1",
+                },
+                {
+                    "type": "caput.tests.test_lint.DoNothing2",
+                    "in": "out2",
+                    "out": "out3",
+                    "requires": "out1",
+                    "params": {
+                        "a_list": [1],
+                    },
+                },
+            ]
+        }
+    }
+
```

Comment:
What happens if we have a task which has a requirement (i.e. defines a `.setup(required_argument)`, but there isn't a `requires` key for the task in the YAML file. Same goes for `process`/`next` methods and `in` keys I guess.

At the very least it's worth testing this and trying to ensure there's a decent warning for it.

Reply:
I tested it, it produces this :+1: 
```
caput.pipeline.PipelineConfigError: Setting up task 2 caused an error - Setting up keys for task QueryDatabase caused an error - Didn't get enough 'requires' keys. Expected at least 1 and only got 0.
```